Add docs for Data Integrity Repair tool (#275)

anvit opened the pull request:
Add documentation for the CLI Data Integrity and Repair tool

anvit committed:
Add docs for Data Integrity Repair tool (#275)

Add documentation for the CLI Data Integrity and Repair tool.
Also removing link to outdated tuning page which has been removed.

melaniekung commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,48 @@ terminal.
+user wants to save this in a different location.
+
+The ``--application``, ``--env``, and ``connection`` options **should not be
+used** - AtoM requires the uses of the pre-set defaults for Symfony to be
```
i think this should say `AtoM requires the *use* of the pre-set defaults`.

anvit replied:
Done!

melaniekung reviewed (approved):
small grammatical suggestion but otherwise LGTM!

fiver-watson commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,48 @@ terminal.
+used** - AtoM requires the uses of the pre-set defaults for Symfony to be
+able to execute the task.
+
+The data integrity repair tool has 3 modes. By default it only generate reports,
```
Not super necessary / kind of implied, but could be good to be explicit here that the default report mode will not make any changes to the data, and is safe to run for information only. I would also suggest that for the other two options, we recommend that users make a database backup first (linking to [here](https://www.accesstomemory.org/docs/latest/admin-manual/maintenance/common-atom-queries/#backing-up-the-database) or similar) - even though the tool _shouldn't_ break anything, any time backend operations are performed on corrupted data, it's always good to be cautious.

fiver-watson commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,48 @@ terminal.
+Find data integrity issues
+==========================
+
+Tries to find data integrity issues in the AtoM database, and attempts to repair
```
Optional, but: could be good to provide a very brief, high-level overview of what data corruption in AtoM is and why it can occur (e.g. long-running operations timing out before completing, leaving incomplete data rows across the AtoM db).... just a thought. 

There is some similar text that could be borrowed from here: 

* https://www.accesstomemory.org/docs/latest/admin-manual/maintenance/troubleshooting/#dealing-with-data-corruption

fiver-watson commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,48 @@ terminal.
+
+   php symfony tools:data-integrity-repair file/path/to/report.csv --mode=delete
+
+:ref:`Back to top <maintenance-cli-tools>`
```
Might be good to have a SEEALSO admonition here, linking to the [Dealing wtih data corruption](https://www.accesstomemory.org/docs/latest/admin-manual/maintenance/troubleshooting/#dealing-with-data-corruption) section of the Troubleshooting page? Or at least the Troubleshooting page in general. Could possibly also link to the generate slugs task, the build nested set task, etc...

fiver-watson reviewed (commented):
hi hi! saw this update and wanted to add a few optional suggestions for you...

In general, I think this PR should also include an update to the [Dealing with data corruption](https://www.accesstomemory.org/docs/latest/admin-manual/maintenance/troubleshooting/#dealing-with-data-corruption) section of the Troubleshooting docs. 

Previously we had a manual SQL query - now we have this task. While the SQL can / should be kept on the Troubleshooting page, anyone finding that section should be made aware of the CLI task first, as it is much easier (and quite possibly safer, for inexperienced SQL users) to run than a SQL query against a production DB. I would suggest adding mention of it in that sections introduction?

I have also added a few optional suggestions here for consideration

anvit committed:
Add relevant links to the data corruption page

Added a link to the CLI data integrity task on the data corruption page.
Also added an explanation about data corruption to the data integrity
task, and added a SEEALSO section to the data integrity task with relevant
links.

anvit commented:
@fiver-watson thanks so much for the feedback! I've updated the PR addressing it.

fiver-watson commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,73 @@ terminal.
+
+The data integrity repair tool has 3 modes. By default it only generates
+a report listing records that have problems and doesn't make any changes to
+them. It is recommended to this first and look through the report before using
```
Missed a word in there - It is recommended to [DO] this first 

or: try, attempt, run, etc.

fiver-watson reviewed (approved):
Thanks for making those changes @anvit ! I found just one mini typo / missed word, but otherwise i think this looks great.

melaniekung commented on admin-manual/maintenance/cli-tools.rst:
```diff
@@ -2728,6 +2728,73 @@ terminal.
+``affected-records.csv`` but a custom filename and path can be specified if the
+user wants to save this in a different location.
+
+The ``--application``, ``--env``, and ``connection`` options **should not be
```
should this say `--connection`?

anvit committed:
Fixed typos with the --connection option

amayita reviewed (approved):
This is AWESOME!!!
